### PR TITLE
Remove related links and migrate existing data to Multi URL Picker

### PR DESCRIPTION
**Pull request by nul800sebastiaan**
This time we'll do it politely: not just remove RelatedLinks but also migrate all the existing data to the awesome new Multi URL Picker!

I've tested many scenarios for the old data
- Old link is an int id
- Old link is a Udi
- Old link is empty
- Old URL is empty
- Old Caption is empty

Everything seems to work beautifully, existing datatypes with the RelatedLinks editor get switched over to the MultiUrlPicker editor, and we don't get any YSODs for missing editors like the last time I tried this. 

What a beauty:
image

In unrelated news: writing migrations for v8 is very nice indeed! 🎉

### CODE REVIEW

**Comment by nul800sebastiaan** [comment]
Ps. Just to make sure: don't merge this if the unit tests don't pass, it will make Stephane unhappy. :)

**Comment by warrenbuckley** [comment]
Code reads fine & I have tested this and all works as expected. Good stuff @nul800sebastiaan

File: src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/ConvertRelatedLinksToMultiUrlPicker.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            var dataTypeIds = dataTypes.Select(x => x.NodeId).ToList();
+
+            var sqlPropertyTpes = Sql()
+                .Select<PropertyTypeDto>()
+                .From<PropertyTypeDto>()
+                .Where<PropertyTypeDto>(x => dataTypeIds.Contains(x.DataTypeId));
+
+            var propertyTypeIds = Database.Fetch<PropertyTypeDto>(sqlPropertyTpes).Select(x => x.Id);
+
+            var sqlPropertyData = Sql()
```
Comment (Shazwazza) [inline src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/ConvertRelatedLinksToMultiUrlPicker.cs]: This SQL call causes an exception if `propertyTypeIds` is an empty array, a count check needs to be added before this call

Comment (Shazwazza) [reply src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/ConvertRelatedLinksToMultiUrlPicker.cs]: I'll fix in v8 branch now

Comment (nul800sebastiaan) [reply src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/ConvertRelatedLinksToMultiUrlPicker.cs]: Duh! Yep, can just exit here if there's no IDs found, thanks!

**Pull request merged**